Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Find.swift

```diff
@@ -404,16 +403,24 @@ extension PathHierarchy.DisambiguationContainer {
         case lookupCollision([(node: PathHierarchy.Node, disambiguation: String)])
     }
     
-    fileprivate func find(_ kind: Substring?, _ hash: Substring?) throws -> PathHierarchy.Node? {
-        try find(kind.map(String.init), hash.map(String.init))
-    }
     /// Attempts to find a value in the disambiguation tree based on partial disambiguation information.
     ///
     /// There are 3 possible results:
     ///  - No match is found; indicated by a `nil` return value.
     ///  - Exactly one match is found; indicated by a non-nil return value.
     ///  - More than one match is found; indicated by a raised error listing the matches and their missing disambiguation.
-    func find(_ kind: String?, _ hash: String?) throws -> PathHierarchy.Node? {
+    func find(_ disambiguation: PathHierarchy.PathComponent.Disambiguation?) throws -> PathHierarchy.Node? {
+        var kind: String?
+        var hash: String?
+        switch disambiguation {
+        case .kindAndHash(kind: let maybeKind, hash: let maybeHash):
+            kind = maybeKind.map(String.init)
+            hash = maybeHash.map(String.init)
+        case nil:
+            kind = nil
+            hash = nil
+        }
```

Comment:
You could move this snippet from here into `PathHierarchy.PathComponent.Disambiguation` and separate it into two calculated properties. Then I think you could simplify this function even more.

Reply:
I've already removed this code in the follow up change—for which I haven't opened a PR yet—that updates how the disambiguated elements are stored in the path hierarchy.